InformationScripting/src/queries/RuntimeQuery.h

```diff
@@ -41,7 +43,9 @@ class INFORMATIONSCRIPTING_API RuntimeQuery : public LinearQuery
 	private:
 		friend class QueryRegistry;
 
-		RuntimeQuery(Model::Node* target, QStringList);
+		QueryExecutor* executor_{};
+
+		RuntimeQuery(Model::Node* target, QStringList, QueryExecutor* executor = nullptr);
```

Comment:
Yes it would be better but if you look at: QueryRegistry::registerQuery  (https://github.com/dimitar-asenov/Envision/pull/191/files#diff-c6b1741a8add6502996a2ba771a07d9aR85 ) is implemented you see that it needs both constructors. There is no static_if so it has to work for both cases.